backend/src/main/java/gov/cdc/usds/simplereport/api/patient/PatientDataResolver.java

```diff
@@ -50,12 +46,16 @@ public ApiFacility getFacility(Person p) {
 
   private CompletableFuture<PatientPreferences> getPatientPreferences(
       Person person, DataFetchingEnvironment dfe) {
+    return loadFuture(person, dfe, PatientPreferencesDataLoader.KEY);
+  }
+
+  private static <T> CompletableFuture<T> loadFuture(
+      AuditedEntity parentObject, DataFetchingEnvironment dfe, final String key) {
     DataLoaderRegistry registry = ((GraphQLContext) dfe.getContext()).getDataLoaderRegistry();
```

Comment:
I'm feel like there's some canonical reason to use the DataLoaderRegistry that's configured into the DataFetchingEnvironment's context... but if the application has a singleton DataLoaderRegistry, configured by Bean, should we just inject _that_ here instead of pulling it from the DFE and doing this typecast?

Reply:
I wonder if graphql-java was relying on the registry being scoped to a single query? Since it's preloading and caching values for the next level of a query, if the registry is a singleton, could stale values leak across requests?

Reply:
Ooooh that is 100% the problem with this. The DataResolvers _must_ be scoped to the request, which is why `ApiUserAwareGraphQlContextBuilder` implements a bunch of `build()` methods. Reworked in 44d884ed

Reply:
Done in b1e4c417